Fix problem on text-width

yasuo-ozu opened the pull request:
# Problem

When we use multipul columns, the width of header and footer is shrinked to the column's width (as below).

# Reference

<blockquote class="twitter-tweet"><p lang="ja" dir="ltr">columnを複数にした場合、ヘッダーとフッターが左に寄っちゃう <a href="https://t.co/vhtri1D515">pic.twitter.com/vhtri1D515</a></p>&mdash; yasuo_ozu@量子コンピュータエンジニア (@yasuo_ozu) <a href="https://twitter.com/yasuo_ozu/status/1613013896209788928?ref_src=twsrc%5Etfw">January 11, 2023</a></blockquote>

yasuo-ozu committed:
Use individual `context` on header or footer.

It solves the text-width problem in header, which occured when we use
multi columns.

yasuo-ozu commented:
# Limitations

When the page has some floating box, the origin point of all the columns are affect by the floating size.
I think, it is the limitation of SATySFi's `page-break-multicolumn` primitive.

abenori commented:
Thank you! Why did you introduce `header-footer-context`? It seems that we don't need this for fixing this bug.

yasuo-ozu commented:
Thanks for comment.
Before this PR, the context of document and header (or footer) is commonalized, so the header and content should have same `text-width`.
In this PR, I split the context for content and {header, footer}. After this PR, the user with lost the way to set the context of header, so I added new config `header-footer-context`.
But I don't think the name is so suitable fot this purpose.

abenori commented:
> Before this PR, the context of document and header (or footer) is commonalized, so the header and content should have same `text-width`.

That sounds not true. I think changing the argument to `get-standard-context`  is sufficient as in your PR. So you don't need `header-footer-context`?

abenori commented:
I guess we don't have to introduce a new interface `header-footer-context`, so I only use half of your code [3ca206c](https://github.com/abenori/satysfi-class-jlreq/commit/3ca206c490877dfe202c73922378afaf5d3c650c). Thank you very much.
